vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -612,59 +652,55 @@ def _transform_value(self, instance, counter_id, value):
-                                                 intervalId=mor['interval'],
-                                                 format='normal')
-        results = perfManager.QueryPerf(querySpec=[query])
+        results = perfManager.QueryPerf(query_specs)
         if results:
-            for result in results[0].value:
-                if result.id.counterId not in self.metrics_metadata[i_key]:
-                    self.log.debug("Skipping this metric value, because there is no metadata about it")
-                    continue
-
-                # Metric types are absolute, delta, and rate
-                try:
-                    metric_name = self.metrics_metadata[i_key][result.id.counterId]['name']
-                except KeyError:
-                    metric_name = None
-
-                if metric_name not in ALL_METRICS:
-                    self.log.debug(u"Skipping unknown `%s` metric.", metric_name)
-                    continue
-
-                if not result.value:
-                    self.log.debug(u"Skipping `%s` metric because the value is empty", metric_name)
-                    continue
-
-                instance_name = result.id.instance or "none"
-                value = self._transform_value(instance, result.id.counterId, result.value[0])
-
-                tags = ['instance:%s' % instance_name]
-                if not mor['hostname']:  # no host tags available
-                    tags.extend(mor['tags'])
-
-                # vsphere "rates" should be submitted as gauges (rate is
-                # precomputed).
-                self.gauge(
-                    "vsphere.%s" % metric_name,
-                    value,
-                    hostname=mor['hostname'],
-                    tags=['instance:%s' % instance_name] + custom_tags
-                )
+            for mor_perfs in results:
+                mor_name = str(mor_perfs.entity)
+                mor = self.morlist[i_key][mor_name]
+                for result in mor_perfs.value:
+                    if result.id.counterId not in self.metrics_metadata[i_key]:
+                        self.log.debug("Skipping this metric value, because there is no metadata about it")
+                        continue
+
+                    # Metric types are absolute, delta, and rate
+                    try:
+                        metric_name = self.metrics_metadata[i_key][result.id.counterId]['name']
+                    except KeyError:
+                        metric_name = None
+
+                    if metric_name not in ALL_METRICS:
+                        self.log.debug(u"Skipping unknown `%s` metric.", metric_name)
+                        continue
+
+                    if not result.value:
+                        self.log.debug(u"Skipping `%s` metric because the value is empty", metric_name)
+                        continue
+
+                    instance_name = result.id.instance or "none"
```

Comment:
is `instance_name ` none a valid value?

Reply:
It is used as a tag value, so we can't change that easily

Reply:
ok